pandas/plotting/_matplotlib/converter.py

```diff
@@ -349,6 +348,8 @@ def __init__(self, locator, tz=None, defaultfmt: str = "%Y-%m-%d") -> None:
 class PandasAutoDateLocator(mdates.AutoDateLocator):
     def get_locator(self, dmin, dmax):
         """Pick the best locator based on a distance."""
+        from dateutil.relativedelta import relativedelta
+
         delta = relativedelta(dmax, dmin)
```

Comment:
seems to me that on L357 we could do `tot_sec = (dmax - dmin).total_seconds()`.  Doing that doesn't break any of our tests.  the relativedelta usage was introduced in https://github.com/pandas-dev/pandas/commit/367e98177573e3cabb712622621088cd193bbdec and i dont see any discussion of why total_seconds wouldn't work.

Reply:
Yeah agreed seems `total_seconds()` should work here